src/freenet/clients/http/Toadlet.java

```diff
@@ -66,10 +68,24 @@
 	 * For still being able to figure out the menu entry through which those so-called invisible Toadlets where accessed,
 	 * this function is necessary.
 	 * 
-	 * @return <code>this</code> by default. Override the function to return something else for invisible Toadlets.
+	 * @param context Can be used to decide the return value, for example to check session cookies using {@link SessionManager}.
+	 * @return The result of {@link #showAsToadlet()}, which is <code>this</code> by default. Override this function to return something else for invisible
+	 *         Toadlets as explained above.
+	 */
+	public Toadlet showAsToadlet(ToadletContext context) {
+	    return showAsToadlet();
```

Comment:
Sorry for the delay. Fixed by PR #340 as this one here is closed already.
Thanks :)